Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp

```diff
@@ -325,6 +325,12 @@ void EmulateIMUCursor(std::optional<RotationalState>* state, ControllerEmu::IMUC
   // Avoid having to double dereference
   auto& st = *state;
 
+  if (!imu_ir_group->GetEnabled())
+  {
+    st = std::nullopt;
```

Comment:
Maybe I'm a bit late to the party, but is there a reason `state` isn't just the return value?

Reply:
I've just followed the pattern of similar functions (`EmulateSwing`, `EmulateTilt`, `EmulateCursor`, `EmulateShake`). Monkey see, monkey do :-)

But I'm guessing it's probably because the `state` argument is both input and output.